Fix for syndat_getjtbul code parsing 3-character storm names (GFSv16.3.21)

KateFriedman-NOAA opened the pull request:
# Description

This PR introduces an update to the `sorc/syndat_getjtbul.fd/getjtbul.f` code to resolve a problem parsing 3-character storm names. This update is being included in the GFSv16.3.21 package (with approval by NCO). 

From the developer @Qingfu-Liu:

> The `gfs_atmos_tropcy_qc_reloc` job did not properly parse the vitals info from a raw bulletin with a tab in the storm name field for storm "SIX". The code sorc/syndat_getjtbul.fd/getjtbul.f converting a tab to a space worked well for storms that have at least 4 characters in their names, but does not work as well for storms with 3-character names. This code change will fix the parsing problem for 3-character storms.

Refs #2591

# Type of change
- [x] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics

Update to code that lives within global-workflow branch.

# How has this been tested?

Tested by NCO and developer

KateFriedman-NOAA committed:
Update for syndat_getjtbul code

The gfs_atmos_tropcy_qc_reloc did not properly parse the vitals info from a raw bulletin
with a tab in the storm name field for storm "SIX". The code sorc/syndat_getjtbul.fd/getjtbul.d
converting a tab to a space worked well for storms that have at least 4 characters in their names,
but does not work as well for storms with 3-character names. This code change will fix the
parsing problem for 3-character storms.

Refs #2591

DavidHuber-NOAA commented on sorc/syndat_getjtbul.fd/getjtbul.f:
```diff
@@ -104,13 +104,21 @@ PROGRAM SYNDAT_GETJTBUL
+            DUMY2K(20:80) = INL(17:77)
+            INL= DUMY2K
+            PRINT *, ' '
+            PRINT *, '==> This is an new-format record with a 4-digit '
```
Minor note.  This should probably read

```fortran
            PRINT *, '==> This is a new-format record with a 4-digit year'
```

KateFriedman-NOAA replied:
Thanks for the comment @DavidHuber-NOAA ! @Qingfu-Liu shall I make the suggested change from Dave?

DavidHuber-NOAA reviewed (approved):
Looks good.  Just a comment on a print statement, but approving as-is.

Qingfu-Liu commented:
Kate,
   You can change the comment from:
PRINT *, '==> This is an new-format record with a 4-digit '
to 
PRINT *, '==> This is a new-format record with a 4-digit year'
Thanks.

KateFriedman-NOAA commented:
> Kate, You can change the comment from: PRINT *, '==> This is an new-format record with a 4-digit ' to PRINT *, '==> This is a new-format record with a 4-digit year' Thanks.

@Qingfu-Liu My terminal shows that adding "year" to the print statement hits the fortan character limit of 72 characters per line. I added the "year" text as a line continuation. It's been a while since I modified fortran so please see the following git diff and let me know if it looks correct? I added "year" to several print statements that needed it:
```
kate.friedman@clogin09:/lfs/h2/emc/global/noscrub/kate.friedman/git/release-gfs.v16.3.21_fork> git diff
diff --git a/sorc/syndat_getjtbul.fd/getjtbul.f b/sorc/syndat_getjtbul.fd/getjtbul.f
index 69666b1e..dc0f8486 100755
--- a/sorc/syndat_getjtbul.fd/getjtbul.f
+++ b/sorc/syndat_getjtbul.fd/getjtbul.f
@@ -116,7 +116,8 @@ c      THE "WINDOWING" TECHNIQUE SINCE SUBSEQUENT LOGIC EXPECTS THIS
             DUMY2K(20:80) = INL(17:77)
             INL= DUMY2K
             PRINT *, ' '
-            PRINT *, '==> This is an new-format record with a 4-digit '
+            PRINT *, '==> This is an new-format record with a 4-digit ',
+     $       'year '
             PRINT *, ' '
           ELSE IF(INL(18:19).EQ.'20') THEN
             DUMY2K(1:17) = INL(1:17)
@@ -124,7 +125,8 @@ c      THE "WINDOWING" TECHNIQUE SINCE SUBSEQUENT LOGIC EXPECTS THIS
             DUMY2K(20:80) = INL(18:78)
             INL= DUMY2K
             PRINT *, ' '
-            PRINT *, '==> This is an new-format record with a 4-digit '
+            PRINT *, '==> This is an new-format record with a 4-digit ',
+     $       'year '
             PRINT *, ' '
           ELSE IF(INL(19:20).EQ.'20') THEN
             DUMY2K(1:18) = INL(1:18)
@@ -132,7 +134,8 @@ c      THE "WINDOWING" TECHNIQUE SINCE SUBSEQUENT LOGIC EXPECTS THIS
             DUMY2K(20:80) = INL(19:79)
             INL= DUMY2K
             PRINT *, ' '
-            PRINT *, '==> This is an new-format record with a 4-digit '
+            PRINT *, '==> This is an new-format record with a 4-digit ',
+     $       'year '
             PRINT *, ' '
```

Qingfu-Liu commented:
Hi Kate,
    please copy the file from this directory:
Cactus: /lfs/h2/emc/physics/noscrub/qingfu.liu/for_NCO_2025/gfs.v16.3.21/sorc/syndat_getjtbul.fd/getjtbul.f
I changed the comments.
Thanks

KateFriedman-NOAA committed:
Update comments related to number of year digits

Refs #2591

KateFriedman-NOAA commented:
> Hi Kate, please copy the file from this directory: Cactus: /lfs/h2/emc/physics/noscrub/qingfu.liu/for_NCO_2025/gfs.v16.3.21/sorc/syndat_getjtbul.fd/getjtbul.f I changed the comments. Thanks

@Qingfu-Liu Even better, thanks! I committed those comment changes @ 1cee9a0. I will now proceed with merging this PR, cutting a new hand-off tag, and letting NCO. Thanks!
